Sources/StreamChat/Repositories/SyncOperations.swift

```diff
@@ -72,12 +72,6 @@ final class WatchChannelOperation: AsyncOperation {
                 return
             }
 
-            // Reset only if it needs recovery
-            if let cid = controller.cid, context.synchedChannelIds.contains(cid) {
-                done(.continue)
-                return
-            }
```

Comment:
Are we sure about this?

Reply:
@nuno-vieira this is a fix that's meant to be safe given the error we are fixing, but maybe not the "most performant" as it can lead to unnecessary calls. (1 more call at most)

There is a task in JIRA to review this in depth and prevent that unnecessary call if possible